src/translate.py

```diff
@@ -3473,7 +3482,7 @@ def handle_rlwinm(
     if right_mask == 0:
         lower_bits = None
     else:
-        lower_bits = BinaryOp.u32(left=source, op=">>", right=Literal(right_shift))
+        lower_bits = BinaryOp.int(left=source, op=">>", right=Literal(right_shift))
```

Comment:
u32 does make sense here, to indicate that it's a logical shift. Only for the lhs though

Reply:
Yeah, totally right! Brain fart on my part.